environment/src/main/java/com/sequenceiq/environment/environment/poller/StackPollerProvider.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.environment.environment.poller;
+
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.dyngr.core.AttemptState;
+import com.sequenceiq.environment.environment.service.stack.StackService;
+import com.sequenceiq.flow.core.FlowConstants;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowLogDBService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.ws.rs.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StackPollerProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackPollerProvider.class);
+
+    private final StackService stackService;
+
+    private final FlowLogDBService flowLogDBService;
+
+    public StackPollerProvider(
+        StackService stackService,
+        FlowLogDBService flowLogDBService) {
+        this.stackService = stackService;
+        this.flowLogDBService = flowLogDBService;
+    }
+
+    public AttemptMaker<Void> stackUpdateConfigPoller(List<String> stackNames, Long envId, String flowId) {
+        List<String> mutableNamesList = new ArrayList<>(stackNames);
+        return () -> {
+            Optional<FlowLog> flowLog = flowLogDBService.getLastFlowLog(flowId);
+            if (flowLog.isPresent() && flowLog.get().getCurrentState().equals(FlowConstants.CANCELLED_STATE)) {
+                return AttemptResults.finishWith(null);
+            }
+            List<String> remaining = new ArrayList<>();
+            List<AttemptResult<Void>> results = collectStackUpdateConfigResults(mutableNamesList,
+                remaining, envId);
+            mutableNamesList.retainAll(remaining);
+            return evaluateResult(results);
+        };
+    }
+
+    private List<AttemptResult<Void>> collectStackUpdateConfigResults(List<String> stackNames,
+        List<String> remaining, Long envId) {
+        return stackNames.stream()
+            .map(stackName -> fetchStackUpdateConfigResults(remaining, stackName, envId))
+            .collect(Collectors.toList());
+    }
+
+    private AttemptResult<Void> fetchStackUpdateConfigResults(List<String> remainingStacks,
+        String stackName, Long envId) {
+        try {
+            stackService.triggerConfigUpdateForStack(stackName);
```

Comment:
why don't we use crn here? I think crn provides some additional uniqueness

Reply:
This simple fronts the CB core stackV4Endpoint.updatePillarConfigurationByName which is 
name based and the CB core pattern for all of these is name based.   

Do we want to start shifting to CRN based for those APIs?

Reply:
so inter service communication should use crn. see @horadla23 comment on cb developer channel on slack:

> Adam Laszlo Horvath  15:19
> I have a request to everyone who is designing, adding, modifying any APIs in CB code. If it is possible, please use CRN of a resource instead of name, since generally CRN is used in the whole platform to identify a resource and also during authorization it requires additional database query to find out the CRN based on the name, because we need CRN to authorize a call in UMS.

Reply:
K done